### DISCUSSION TITLE
Displaying Bar Chart UI

### DISCUSSION THREAD

**Discussion by hci-ikahera** [Budibase/budibase#10130] [Help] [answered]
Hello everyone,
Anyone here know how to fix this?
I have set the height using defined value but the width is auto.
But on the first load, the width does not stretch. Everytime I try to reload it, it keeps changing the width.
Image #1 -> Display on the first load.
Image #2 -> Display after I try to reload it.
Image #3 -> Display after I execute Refresh Provider.

Image #1
<img width="852" alt="Screenshot 2023-03-28 at 06 05 21" src="https://user-images.githubusercontent.com/95213000/228134858-836d2180-c991-414b-a47f-9240414a8ee0.png">

Image #2
<img width="844" alt="Screenshot 2023-03-28 at 06 05 37" src="https://user-images.githubusercontent.com/95213000/228134881-a55ba253-0f7f-4620-9d28-0738c08c4fd4.png">

Image #3
<img width="854" alt="Screenshot 2023-03-28 at 06 05 50" src="https://user-images.githubusercontent.com/95213000/228134894-e50d7231-e6f8-4bbd-a07e-a174d1271ed5.png">

Thank you for any kind of help.

Regards,
Ika

**Comment by andz-bb** [comment, marked as answer]
Hey @hci-ikahera 

I've seen issues like this before with how the chart components handle sizing in Budibase. One issue I've raised that comes to mind is this one: #9462 - when I was testing it a few months ago I saw similar issues with chart sizing/responsiveness to what you're seeing here.

When I tried on a fresh app I was unable to recreate with refreshing the data provider. Could you provide a bit more context about how your screen is set up, and let me know if there are any errors in your browser's developer tools console? Alternatively feel free to send your app export to support@budibase.com or to me (Andrew T#0001) on [our Discord server](https://discord.gg/ZepTmGbtfF) and I can have a look for fixes/workarounds. Thanks!

**Comment by hci-ikahera** [comment]
I just found it already worked the way it should. Thank you 🙏🏼
